src/expandable-section/expandable-section-header.tsx

```diff
@@ -183,19 +182,29 @@ const ExpandableHeaderTextWrapper = ({
           description={description}
           counter={headerCounter}
           info={headerInfo}
-          actions={headerActions}
+          actions={actions}
           headingTagOverride={headingTagOverride}
         >
           {headerButton}
         </InternalHeader>
       ) : (
         <>
-          <HeadingTag
-            className={clsx(styles['header-wrapper'], headingTagListeners && styles['click-target'])}
-            {...headingTagListeners}
-          >
-            {headerButton}
-          </HeadingTag>
+          {actions ? (
+            <div className={clsx(styles['header-wrapper'], styles['header-wrapper-with-actions'])}>
+              <HeadingTag className={styles['click-target']} {...headingTagListeners}>
```

Comment:
Why is the `click-target` class added here without conditions but in the same code below it has a condition: `headingTagListeners && styles['click-target']`

Is there a potential for a bug? Any chances to reduce duplicated code?

Reply:
Refactored [here](https://github.com/cloudscape-design/components/pull/1644/commits/db54fea422a9bc0f6ae136b85905c5b659dfb85b), with the caveat that now there is an unconditional wrapping div even if there is nothing in the actions slot, but I guess the benefit of code simplicity overrules this?